File: 326-power-of-three/326-power-of-three.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        
        def three(n):
            
            if n==1:
                return True
            elif n and n%3==0:
                return three(n//3)
            else:
                return False
            
        
        return three(n)
        
        

File: 326-power-of-three/326-power-of-three.cs
public class Solution {
    public bool IsPowerOfThree(int n) {
        if (n==1){
            return true;
        }
        else if(n!=0 && n%3==0){
            return this.IsPowerOfThree(n/3);
        }
        else{
            return false;
        }
    }
}